core/src/main/java/com/amplifyframework/api/graphql/error/AppSyncExtensions.java

```diff
@@ -0,0 +1,72 @@
+/*
+ * Copyright 2020 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+package com.amplifyframework.api.graphql.error;
+
+import androidx.annotation.Nullable;
+
+import com.amplifyframework.util.Immutable;
+
+import java.util.Map;
+
+public final class AppSyncExtensions {
```

Comment:
It'd be good to avoid putting this in the `core` module, if possible. Can we put it under `aws-api`, or `aws-datastore` modules, instead?

Reply:
I initially put it in `core` since [that's the current approach on iOS](https://github.com/aws-amplify/amplify-ios/pull/368/files#diff-2eab2075f7c39215e0895d8018aed92a) (though this PR is pending refactoring) and I understand we don't really want AppSync specific stuff in core.

AppSyncExtensions is really just a helper object with a couple potential consumers:
(1)  A unit test in `aws-api`
(2) An end user consuming the `aws-api` 
(3) `aws-datastore` could use it to parse the error to handle conflict resolution.

Any of the above three use cases has the ability to access extensions via the Map<String, Object> though.

If we put it in `aws-api`, then it would not be accessible by `aws-datastore`.  

I'm sort of inclined to move it to `aws-datastore` because that satisfies (3).  For (1), we can just move the unit test to datastore, and for (2), not worry about it now since the end user use case isn't totally understood, and they can just use the Map<String, Object>.

@jamesonwilliams I don't have much context on datastore yet, so you tell me, does it make sense to put it in `aws-datastore`?

Reply:
Hey @richardmcclellan, yea, I think `aws-datastore` should work fine, and be an improvement over `core`.

We have App Sync details in both `aws-api`, and `aws-datastore`, at the moment, with no _real_ rules about how or why.

When we get some time, I think it probably makes sense to create a new module called `appsync`. `aws-api` and `aws-datastore` would impelment the contracts in `core`, by means of leveraging a common `appsync` library module.